### PR TITLE
fixing typos

### PR DESCRIPTION
**Pull request by ZmagoD**

### CODE REVIEW

File: app/controllers/api/v1/sessions_controller.rb [hunk 1]
```diff
@@ -13,4 +13,21 @@ def create
...
+    # respond_to_on_destroy 
+    # binding.pry
+    # user = User.find_by(authentication_token: params[:authentication_token])
+    # user.authentication_token = nil
+    # user.save
+    
+    user = User.find_by_authentication_token(request.headers['X-API-TOKEN'])
+    if user
+      user.authentication_token=nil
+      render :json => { :message => 'Session deleted.' }, :success => true, :status => 204
```
Comment (tochman) [inline app/controllers/api/v1/sessions_controller.rb]: I always try to use the new Hash format
```
render json: { message: 'Session deleted.' }, success:  true, status: 204
```

Comment (tochman) [inline app/controllers/api/v1/sessions_controller.rb]: Another question, why `:success => true` ?

File: spec/requests/api/v1/sessions_spec.rb [hunk 2]
```diff
@@ -39,4 +39,15 @@
     end
 
   end
+  
+  describe 'DELETE /api/v1/users/sign_out' do
+    describe 'user log out' do
+      before(:each) do
+        post '/api/v1/users/sign_in', {user:{email:"#{user.email}", password:"#{user.password}"}}, headers
+        
+        delete '/api/v1/users/sign_out', {user:{authentication_token: "#{user.authentication_token}"}}, headers
```
Comment (tochman) [inline spec/requests/api/v1/sessions_spec.rb]: Will we be able to do a `DELETE` request from our Angular app?

File: spec/requests/api/v1/sessions_spec.rb [hunk 3]
```diff
@@ -39,4 +39,15 @@
...
+      end
+      it { should respond_with 204 }
```
Comment (tochman) [inline spec/requests/api/v1/sessions_spec.rb]: I would suggest passing in a block here and make the `DELETE` call in it. 
```
it 'some message....' do
  delete '/api/v1/users/sign_out', {user:{authentication_token: "#{user.authentication_token}"}}, headers
  expect(response.status).to eq 204
end
```

Comment (tochman) [inline spec/requests/api/v1/sessions_spec.rb]: If you use ` it { should respond_with 204 }` you should probably use `is_expected_to`

**Comment by ZmagoD** [comment]
Api::V1::SessionsController DELETE /api/v1/users/sign_out user log out should set the user authentication token to nil
     Failure/Error: user = User.find_by_authentication_token(request.headers['X-API-TOKEN'])
     
     NoMethodError:
       undefined method `find_by_authentication_token' for Godmin::Authentication::User:Module

**Comment by ZmagoD** [comment]
This come when I run the spec

**Comment by tochman** [comment]
Have a look at my solution for the Facebook method.

**Comment by ZmagoD** [comment]
I hadn't done many api authentication but the thing here is getting smelly :)

**Comment by tochman** [comment]
I know. My solution is rather hacky. The problem is the Godmin modules. We can look for a better solution further down the line and refactor it later. I can talk to the creators of the Godmin gem perhaps and ask them for input.

**Comment by ZmagoD** [comment]
i will hold his arms and you will be beating him :)

**Comment by tochman** [comment]
Haha! Yeah, I can go ahead and merge this and fix that Godmin interference if you want. Also, I'm thinking about the DELETE method, we should pass the token using headers rather than as part of the request body.

**Comment by ZmagoD** [comment]
OK

**Pull request merged**